nbs/13_cog_sights.ipynb

```diff
@@ -0,0 +1,426 @@
+    "    12: \"crop\",\n",
+    "    13: \"built\",\n",
+    "    14: \"non-natural tree cover\",\n",
+    "    15: \"non-natural short vegetation\",\n",
+    "    16: \"non-natural water\",\n",
+    "    17: \"wet non-natural tree cover\",\n",
+    "    18: \"non-natural peat tree cover\",\n",
+    "    19: \"wet non-natural short vegetation\",\n",
+    "    20: \"non-natural peat short vegetation\",\n",
+    "    21: \"non-natural bare\"\n",
+    "}\n",
+    "\n",
+    "DEFAULT_COG_PATH = \"s3://gfw-data-lake/umd_glad_dist_alerts/v20250329/raster/epsg-4326/cog/default.tif\"\n",
+    "INTENSITY_COG_PATH = \"s3://gfw-data-lake/umd_glad_dist_alerts/v20250329/raster/epsg-4326/cog/intensity.tif\"\n",
+    "STAC_API_URL = \"https://eoapi.zeno-staging.ds.io/stac\"\n",
+    "AWS_PROFILE = \"zeno_internal_sso\"\n",
+    "REFERENCE_DATE = datetime(2015, 1, 1)\n",
+    "\n",
+    "def convert_date_range_to_days(date_range):\n",
+    "    start_date = datetime.strptime(str(date_range[0])[:10], \"%Y-%m-%d\")\n",
+    "    end_date = datetime.strptime(str(date_range[1])[:10], \"%Y-%m-%d\")\n",
+    "    return (start_date - REFERENCE_DATE).days, (end_date - REFERENCE_DATE).days, start_date, end_date\n",
+    "\n",
+    "def load_alert_data(geometry):\n",
+    "    session = boto3.Session(profile_name=AWS_PROFILE)\n",
+    "    with rasterio.Env(AWSSession(session), AWS_REQUEST_PAYER=\"requester\"):\n",
+    "        with rasterio.open(DEFAULT_COG_PATH) as src1:\n",
+    "            default_data, _ = mask(src1, geometry, crop=True)\n",
+    "        with rasterio.open(INTENSITY_COG_PATH) as src2:\n",
+    "            intensity_data, _ = mask(src2, geometry, crop=True)\n",
+    "    return default_data[0], intensity_data[0]\n",
+    "\n",
+    "def load_natural_lands_mosaic(aoi_geom, start_date, end_date):\n",
+    "    stac = Client.open(STAC_API_URL)\n",
+    "    search = stac.search(\n",
+    "        collections=[\"natural-lands-map-v1-1\"],\n",
+    "        intersects=aoi_geom,\n",
+    "        datetime=f\"{start_date.date().isoformat()}/{end_date.date().isoformat()}\",\n",
+    "        max_items=50,\n",
+    "    )\n",
+    "    items = list(search.get_items())\n",
+    "    if not items:\n",
+    "        raise ValueError(\"No Natural Lands items found for AOI.\")\n",
+    "\n",
+    "    print(f\"Found {len(items)} items\")\n",
+    "\n",
+    "    da = stack(\n",
+    "        items,\n",
+    "        bounds_latlon=aoi_geom.bounds,\n",
+    "        snap_bounds=True,\n",
+    "        epsg=4326\n",
+    "    )\n",
+    "    \n",
+    "    # Apply chunking\n",
+    "    da = da.chunk({\"x\": 1024, \"y\": 1024})\n",
+    "\n",
+    "\n",
+    "    # Collapse time dimension if multiple timestamps\n",
+    "    if \"time\" in da.dims:\n",
+    "        da = da.astype(\"int16\").max(\"time\")\n",
+    "\n",
+    "    return da.squeeze()\n",
+    "\n",
+    "\n",
+    "#@tool\n",
+    "def filter_and_count_pixels(state: Dict) -> Dict:\n",
```

Comment:
This tool makes sense to me, I think we'll need a tool dedicated to DIST alerts because of the custom date filter logic.